### PR TITLE
Adding a Kotlin meme-id module

### PR DESCRIPTION
**Pull request by ahinchman1**
Added a `memeid-kotlin` module and the bare minimum of being able to compile Kotlin code.

### CODE REVIEW

File: memeid-kotlin/src/test/kotlin/memeid/kotlin/UUIDTest.kt [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+      |  From UUID: $fromUUID | version: ${fromUUID.version()} | variant: ${fromUUID.variant()}
+      |  From significant bits: $fromSigBits | version: ${fromSigBits.version()} | variant: ${fromSigBits.variant()}
+      |  From String: $fromString | version: ${fromString?.version()} | variant: ${fromString?.variant()}
+      --------------------------------------------------------------------""".trimIndent())
+
+    assertThat(fromUUID.isV1, equalTo(true))
+    // assertThat(fromSigBits.isV1, equalTo(true))   <-- returns a NIL version
+    assertThat(fromString?.isV1, equalTo(true))
+  }
+
```
Comment (ahinchman1) [inline memeid-kotlin/src/test/kotlin/memeid/kotlin/UUIDTest.kt:58]:
```
Generated Version 1 UUID: c7e2d512-aa62-11ea-b22b-b9f29368240d | UUID Version: 1 | UUID Variant: 1 
|  From UUID: c7e2d512-aa62-11ea-b22b-b9f29368240d | version: 1 | variant: 2
|  From significant bits: b22bb9f2-9368-240d-c7e2-d512aa6211ea | version: 2 | variant: 6
|  From String: c7e2d512-aa62-11ea-b22b-b9f29368240d | version: 1 | variant: 2
--------------------------------------------------------------------
Generated Version 1 UUID: c7e2d513-aa62-11ea-b22b-5743c64e3fcd | UUID Version: 1 | UUID Variant: 1 
|  From UUID: c7e2d513-aa62-11ea-b22b-5743c64e3fcd | version: 1 | variant: 2
|  From significant bits: b22b5743-c64e-3fcd-c7e2-d513aa6211ea | version: 3 | variant: 6
|  From String: c7e2d513-aa62-11ea-b22b-5743c64e3fcd | version: 1 | variant: 2
--------------------------------------------------------------------
Generated Version 1 UUID: c7e2d514-aa62-11ea-b22b-cdb0cf395bda | UUID Version: 1 | UUID Variant: 1 
|  From UUID: c7e2d514-aa62-11ea-b22b-cdb0cf395bda | version: 1 | variant: 2
|  From significant bits: b22bcdb0-cf39-5bda-c7e2-d514aa6211ea | version: 5 | variant: 6
|  From String: c7e2d514-aa62-11ea-b22b-cdb0cf395bda | version: 1 | variant: 2
--------------------------------------------------------------------
Generated Version 1 UUID: c7e2d515-aa62-11ea-b22b-fdaf8ca4b921 | UUID Version: 1 | UUID Variant: 1 
|  From UUID: c7e2d515-aa62-11ea-b22b-fdaf8ca4b921 | version: 1 | variant: 2
|  From significant bits: b22bfdaf-8ca4-b921-c7e2-d515aa6211ea | version: 11 | variant: 6
|  From String: c7e2d515-aa62-11ea-b22b-fdaf8ca4b921 | version: 1 | variant: 2
--------------------------------------------------------------------
Generated Version 1 UUID: c7e2d516-aa62-11ea-b22b-1f8769412f11 | UUID Version: 1 | UUID Variant: 1 
|  From UUID: c7e2d516-aa62-11ea-b22b-1f8769412f11 | version: 1 | variant: 2
|  From significant bits: b22b1f87-6941-2f11-c7e2-d516aa6211ea | version: 2 | variant: 6
|  From String: c7e2d516-aa62-11ea-b22b-1f8769412f11 | version: 1 | variant: 2
--------------------------------------------------------------------
Generated Version 1 UUID: c7e2d517-aa62-11ea-b22b-edff3dd81d30 | UUID Version: 1 | UUID Variant: 1 
|  From UUID: c7e2d517-aa62-11ea-b22b-edff3dd81d30 | version: 1 | variant: 2
|  From significant bits: b22bedff-3dd8-1d30-c7e2-d517aa6211ea | version: 1 | variant: 6
|  From String: c7e2d517-aa62-11ea-b22b-edff3dd81d30 | version: 1 | variant: 2
--------------------------------------------------------------------
Generated Version 1 UUID: c7e2d518-aa62-11ea-b22b-c1846bf372d0 | UUID Version: 1 | UUID Variant: 1 
|  From UUID: c7e2d518-aa62-11ea-b22b-c1846bf372d0 | version: 1 | variant: 2
|  From significant bits: b22bc184-6bf3-72d0-c7e2-d518aa6211ea | version: 7 | variant: 6
|  From String: c7e2d518-aa62-11ea-b22b-c1846bf372d0 | version: 1 | variant: 2
--------------------------------------------------------------------
Generated Version 1 UUID: c7e2d519-aa62-11ea-b22b-07ba062fa679 | UUID Version: 1 | UUID Variant: 1 
|  From UUID: c7e2d519-aa62-11ea-b22b-07ba062fa679 | version: 1 | variant: 2
|  From significant bits: b22b07ba-062f-a679-c7e2-d519aa6211ea | version: 10 | variant: 6
|  From String: c7e2d519-aa62-11ea-b22b-07ba062fa679 | version: 1 | variant: 2
--------------------------------------------------------------------
Generated Version 1 UUID: c7e2d51a-aa62-11ea-b22b-e9fea9fc990c | UUID Version: 1 | UUID Variant: 1 
|  From UUID: c7e2d51a-aa62-11ea-b22b-e9fea9fc990c | version: 1 | variant: 2
|  From significant bits: b22be9fe-a9fc-990c-c7e2-d51aaa6211ea | version: 9 | variant: 6
|  From String: c7e2d51a-aa62-11ea-b22b-e9fea9fc990c | version: 1 | variant: 2
--------------------------------------------------------------------
Generated Version 1 UUID: c7e2d51b-aa62-11ea-b22b-235276477353 | UUID Version: 1 | UUID Variant: 1 
|  From UUID: c7e2d51b-aa62-11ea-b22b-235276477353 | version: 1 | variant: 2
|  From significant bits: b22b2352-7647-7353-c7e2-d51baa6211ea | version: 7 | variant: 6
|  From String: c7e2d51b-aa62-11ea-b22b-235276477353 | version: 1 | variant: 2
--------------------------------------------------------------------
Generated Version 1 UUID: c7e2d51c-aa62-11ea-b22b-db8445ef4c4f | UUID Version: 1 | UUID Variant: 1 
|  From UUID: c7e2d51c-aa62-11ea-b22b-db8445ef4c4f | version: 1 | variant: 2
|  From significant bits: b22bdb84-45ef-4c4f-c7e2-d51caa6211ea | version: 4 | variant: 6
|  From String: c7e2d51c-aa62-11ea-b22b-db8445ef4c4f | version: 1 | variant: 2
--------------------------------------------------------------------
Generated Version 1 UUID: c7e2d51d-aa62-11ea-b22b-99e4cb07a46e | UUID Version: 1 | UUID Variant: 1 
|  From UUID: c7e2d51d-aa62-11ea-b22b-99e4cb07a46e | version: 1 | variant: 2
|  From significant bits: b22b99e4-cb07-a46e-c7e2-d51daa6211ea | version: 10 | variant: 6
|  From String: c7e2d51d-aa62-11ea-b22b-99e4cb07a46e | version: 1 | variant: 2
--------------------------------------------------------------------
Generated Version 1 UUID: c7e2d51e-aa62-11ea-b22b-65f9c51ad8c9 | UUID Version: 1 | UUID Variant: 1 
|  From UUID: c7e2d51e-aa62-11ea-b22b-65f9c51ad8c9 | version: 1 | variant: 2
|  From significant bits: b22b65f9-c51a-d8c9-c7e2-d51eaa6211ea | version: 13 | variant: 6
|  From String: c7e2d51e-aa62-11ea-b22b-65f9c51ad8c9 | version: 1 | variant: 2
--------------------------------------------------------------------
Generated Version 1 UUID: c7e2d51f-aa62-11ea-b22b-3b50431abc0d | UUID Version: 1 | UUID Variant: 1 
|  From UUID: c7e2d51f-aa62-11ea-b22b-3b50431abc0d | version: 1 | variant: 2
|  From significant bits: b22b3b50-431a-bc0d-c7e2-d51faa6211ea | version: 11 | variant: 6
|  From String: c7e2d51f-aa62-11ea-b22b-3b50431abc0d | version: 1 | variant: 2
--------------------------------------------------------------------
```

File: memeid-kotlin/src/test/kotlin/memeid/kotlin/UUIDTest.kt [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+      |  From UUID: $fromUUID | version: ${fromUUID.version()} | variant: ${fromUUID.variant()}
+      |  From significant bits: $fromSigBits | version: ${fromSigBits.version()} | variant: ${fromSigBits.variant()}
+      |  From String: $fromString | version: ${fromString?.version()} | variant: ${fromString?.variant()}
+      --------------------------------------------------------------------""".trimIndent())
+
+      assertThat(fromUUID.isV4, equalTo(true))
+      // assertThat(fromSigBits.isV4, equalTo(true)) <-- returns version 2
+      assertThat(fromString?.isV4, equalTo(true))
+    }
+  }
```
Comment (ahinchman1) [inline memeid-kotlin/src/test/kotlin/memeid/kotlin/UUIDTest.kt:79]:
```
Generated Version 4 UUID: 9e7bb446-1cf2-4684-bbb4-461cf27684ff | UUID Version: 4 | UUID Variant: 4 
|  From UUID: 9e7bb446-1cf2-4684-bbb4-461cf27684ff | version: 4 | variant: 2
|  From significant bits: bbb4461c-f276-84ff-9e7b-b4461cf24684 | version: 8 | variant: 2
|  From String: 9e7bb446-1cf2-4684-bbb4-461cf27684ff | version: 4 | variant: 2
--------------------------------------------------------------------
Generated Version 4 UUID: 60977aed-d1f4-4927-977a-edd1f4292737 | UUID Version: 4 | UUID Variant: 4 
|  From UUID: 60977aed-d1f4-4927-977a-edd1f4292737 | version: 4 | variant: 2
|  From significant bits: 977aedd1-f429-2737-6097-7aedd1f44927 | version: 2 | variant: 0
|  From String: 60977aed-d1f4-4927-977a-edd1f4292737 | version: 4 | variant: 2
--------------------------------------------------------------------
Generated Version 4 UUID: a14eba6d-6853-4bf7-8eba-6d6853abf763 | UUID Version: 4 | UUID Variant: 4 
|  From UUID: a14eba6d-6853-4bf7-8eba-6d6853abf763 | version: 4 | variant: 2
|  From significant bits: 8eba6d68-53ab-f763-a14e-ba6d68534bf7 | version: 15 | variant: 2
|  From String: a14eba6d-6853-4bf7-8eba-6d6853abf763 | version: 4 | variant: 2
--------------------------------------------------------------------
Generated Version 4 UUID: 0f92a250-9a4e-4938-92a2-509a4e8938cd | UUID Version: 4 | UUID Variant: 4 
|  From UUID: 0f92a250-9a4e-4938-92a2-509a4e8938cd | version: 4 | variant: 2
|  From significant bits: 92a2509a-4e89-38cd-0f92-a2509a4e4938 | version: 3 | variant: 0
|  From String: 0f92a250-9a4e-4938-92a2-509a4e8938cd | version: 4 | variant: 2
--------------------------------------------------------------------
Generated Version 4 UUID: 8f62842d-4ab9-4094-a284-2d4ab9c09423 | UUID Version: 4 | UUID Variant: 4 
|  From UUID: 8f62842d-4ab9-4094-a284-2d4ab9c09423 | version: 4 | variant: 2
|  From significant bits: a2842d4a-b9c0-9423-8f62-842d4ab94094 | version: 9 | variant: 2
|  From String: 8f62842d-4ab9-4094-a284-2d4ab9c09423 | version: 4 | variant: 2
--------------------------------------------------------------------
Generated Version 4 UUID: 79b09618-5c51-4d0a-b096-185c51fd0acc | UUID Version: 4 | UUID Variant: 4 
|  From UUID: 79b09618-5c51-4d0a-b096-185c51fd0acc | version: 4 | variant: 2
|  From significant bits: b096185c-51fd-0acc-79b0-96185c514d0a | version: 0 | variant: 0
|  From String: 79b09618-5c51-4d0a-b096-185c51fd0acc | version: 4 | variant: 2
--------------------------------------------------------------------
Generated Version 4 UUID: 8a923c9e-94cc-4d6d-923c-9e94ccdd6de7 | UUID Version: 4 | UUID Variant: 4 
|  From UUID: 8a923c9e-94cc-4d6d-923c-9e94ccdd6de7 | version: 4 | variant: 2
|  From significant bits: 923c9e94-ccdd-6de7-8a92-3c9e94cc4d6d | version: 6 | variant: 2
|  From String: 8a923c9e-94cc-4d6d-923c-9e94ccdd6de7 | version: 4 | variant: 2
--------------------------------------------------------------------
```

**Comment by ahinchman1** [comment]
@alejandrohdezma hey! did you ever get a chance to look into the CI portion? If not maybe I could?

**Comment by alejandrohdezma** [comment]
> @alejandrohdezma hey! did you ever get a chance to look into the CI portion? If not maybe I could?

Nope! I didn't, take a look at it now :)

**Comment by alejandrohdezma** [comment]
@ahinchman1 okey, I've added a few commits doing some fixes that are necessary for this to work. However, I haven't managed to get tests to work correctly :( It seems that the [kotlin-plugin](https://github.com/pfn/kotlin-plugin) for SBT is a bit outdated and hasn't been updated on most than 6 months...

So the options are either trying to fix the plugin and keep the build SBT-based or create a gradle-based build just for the `memeid-kotlin` module and create new Github Actions accordingly...

**Comment by fedefernandez** [comment]
@alejandrohdezma I think if even we got this working, maintaining this could be a challenge. I'd go for a gradle module instead.

**Comment by ahinchman1** [comment]
Hey, sorry, I just saw this! Yes, I think @fedefernandez has a great suggestion - I can maybe research if there's an sbt plugin to integrate gradle/sbt, or we can move the whole thing to gradle. @alejandrohdezma what do you think?

**Pull request closed without merging**